app/controllers/recommendation_controller.py

```diff
@@ -10,9 +10,22 @@ def handle(self):
         emails = [email.strip() for email in request.args.get('emails').split(',')]
         users = User.query.filter(User.email.in_(emails))
 
-        movie_id_sets = [{mp.external_movie_id for mp in user.movies} for user in users]
-        common_movie_ids = set.intersection(*movie_id_sets) if movie_id_sets else set()
-        recommended_movie_ids = list(common_movie_ids)[:5] # Limit recommendations to 5
+        positive_movie_id_sets = []
+        negative_movie_id_sets = []
+        for user in users:
+            positive_set = set()
+            negative_set = set()
+            for mp in user.movies:
```

Comment:
Hmm this could be
```py
for user in users:
    user_movies = user.movies # only do the query once
    positive_movie_id_sets.append({m for m in user_movies if m.preference_type == PreferenceTypes.positive})
    negative_movie_id_sets.append({m for m in user_movies if m.preference_type == PreferenceTypes.negative})
```